[ONNX] Add ONNX ExportedProgram tests

titaiwangms opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #114633

Fix #114166 
Fix #113705 

This PR references tests from `test_export.py` to make sure the exported program from PyTorch can all be successfully exported into ONNX model.

titaiwangms committed:
[ONNX] Add ONNX ExportedProgram tests

[ghstack-poisoned]

titaiwangms commented on test/onnx/test_fx_to_onnx_with_onnxruntime.py:
```diff
@@ -43,8 +43,14 @@ def _parameterized_class_attrs_and_values():
+            (
+                True,
+                False,
+            ),
```
Set this back

titaiwangms commented on test/onnx/test_fx_to_onnx_with_onnxruntime.py:
```diff
@@ -181,35 +187,33 @@ def func(x, y):
+        # ref_outputs = onnx_program.adapt_torch_outputs_to_onnx(func(tensor_x, 9.0))
+        # _ = onnx_test_common.run_ort(onnx_program, onnx_format_args)
+        # for ref_output, ort_output in zip(ref_outputs, ort_outputs):
+        #     torch.testing.assert_close(ref_output, torch.tensor(ort_output))
```
Revert this back

titaiwangms commented on torch/onnx/_internal/exporter.py:
```diff
@@ -1129,6 +1129,8 @@ def export(self) -> ONNXProgram:
                 self.options, self.model, self.model_args, self.model_kwargs
             )
 
+            graph_module.print_readable()
```
Remove this

titaiwangms commented on torch/onnx/_internal/io_adapter.py:
```diff
@@ -414,6 +414,7 @@ def apply(
                 is not identical to the `SpecTree` output produced from the first
                 `model_outputs` that was passed to this method.
         """
+        # import pdb; pdb.set_trace()
```
Remove this

titaiwangms committed:
Update on "[ONNX] Add ONNX ExportedProgram tests"




[ghstack-poisoned]

thiagocrepaldi commented on test/onnx/torch_export/test_torch_export_with_onnxruntime.py:
```diff
@@ -0,0 +1,298 @@
+
+        # NOTE: Careful with the input format. The input format should be
+        # consistent with how the model is exported.
+        test_inputs = ((torch.randn(4, 4), torch.randn(6, 4)), torch.randn(4, 4))
```
you should use `onnx_program.adapt_torch_inputs_tyo_onnx()` to adapt the "normal torch input" to the format expected by the onnx model

titaiwangms replied:
Not sure about this. The input is adpated inside onnx_program __call__.

thiagocrepaldi commented on test/onnx/torch_export/test_torch_export_with_onnxruntime.py:
```diff
@@ -0,0 +1,298 @@
+
+        # NOTE: A model should be fed with the input formats that
+        # how the model is exported
+        inputs = {"x": torch.randn(6, 3), "y": torch.randn(6, 3)}
```
same as above. use `onnx_program.adapt_torch_inputs_tyo_onnx()`

titaiwangms replied:
The note is kept for torch exported program usage.

titaiwangms committed:
Update on "[ONNX] Add ONNX ExportedProgram tests"


This PR references tests from `test_export.py` to make sure the exported program from PyTorch can all be successfully exported into ONNX model.

[ghstack-poisoned]

titaiwangms commented on test/onnx/torch_export/test_torch_export_with_onnxruntime.py:
```diff
@@ -0,0 +1,298 @@
+        if input_kwargs is None:
+            input_kwargs = {}
+
+        onnx_outputs = onnx_exported_program(*input_args, **input_kwargs)
```
Added

titaiwangms commented on test/onnx/test_fx_to_onnx_with_onnxruntime.py:
```diff
@@ -515,9 +515,14 @@ def forward(self, x):
-        "of node (n0__4) does not match expected type (tensor(int64))"
+    @pytorch_test_common.xfail_if_model_type_is_exportedprogram(
+        "RuntimeError:"
+        " Found following user inputs located at [0] are mutated. This is currently banned in the aot_export workflow."
```
Error message changed

titaiwangms commented:
@pytorchbot merge
